src/formatDate.js

```diff
@@ -8,7 +8,32 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateParts = date.split(fromFormat.pop());
```

Comment:
When splitting the `date` string, it's important to ensure that the separator from `fromFormat` is indeed at the end of the array. If the format ever changes to include the separator at a different position, this code will break. Consider directly accessing the separator using the index of the last element in `fromFormat` for clarity and to avoid mutation of the `fromFormat` array.